src/modules/decklink/consumer_decklink.cpp

```diff
@@ -392,6 +392,7 @@ class DeckLinkConsumer
 			return false;
 		}
 		mlt_properties_set_int( properties, "top_field_first", m_displayMode->GetFieldDominance() == bmdUpperFieldFirst );
+		mlt_properties_set_int( properties, "progressive", m_displayMode->GetFieldDominance() == bmdProgressiveFrame );
```

Comment:
The call to getDisplayMode() already requires the mlt_profile.progressive attribute to match the decklink's mode.